custom/src/main/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabled.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry;
+
+import static com.splunk.opentelemetry.SplunkConfiguration.METRICS_ENABLED_PROPERTY;
+import static com.splunk.opentelemetry.SplunkConfiguration.METRICS_FULL_COMMAND_LINE;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ConditionalResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.util.logging.Logger;
+
+@AutoService(ResourceProvider.class)
+public class TruncateCommandLineWhenMetricsEnabled implements ConditionalResourceProvider {
+
+  private static final Logger logger =
+      Logger.getLogger(TruncateCommandLineWhenMetricsEnabled.class.getName());
+
+  private String commandLine;
```

Comment:
Keeping state this way seems kinda hacky. Instead of using the `ConditionalResourceProvider`, how about making it a `AutoConfigurationCustomizerProvider` and calling `AutoConfigurationCustomizer#addResourceCustomizer()`?

Also, using the resource customizer has one more advantage: it receives the final (well, almost final) version of the resource, that also includes attributes from the system props/env vars.

Reply:
Agreed, I hate this. If the resource customizer will work, that sounds better. Lemme try it. THANKS!

Reply:
Looks good, thanks!